src/sagemaker/workflow/_repack_model.py

```diff
@@ -39,14 +39,14 @@ def repack(inference_script, model_archive, dependencies=None, source_dir=None):
 
     Args:
         inference_script (str): The path to the custom entry point.
-        model_archive (str): The name of the model TAR archive.
+        model_archive (str): The name or path (e.g. s3 uri) of the model TAR archive.
         dependencies (str): A space-delimited string of paths to custom dependencies.
         source_dir (str): The path to a custom source directory.
     """
 
     # the data directory contains a model archive generated by a previous training job
     data_directory = "/opt/ml/input/data/training"
-    model_path = os.path.join(data_directory, model_archive)
+    model_path = os.path.join(data_directory, model_archive.split("/")[-1])
 
```

Comment:
Why are we changing this?

Reply:
This `_repack_model.py` is a script which will be uploaded and executed during Pipeline execution runtime (backend), which means all pipeline variables are correctly parsed to proper type like str at that time. 
So it's safe to do the str split in this script.
A corresponding change is in `_utils.py` below ([link](https://github.com/aws/sagemaker-python-sdk/pull/3043/commits/b1a530aacc41610fefcae0f75e34dc8586220cba#diff-10872c4addf3d40c31388b81648eaa126c1354afbcd3ec7d183b8b9207a81b83L137-L142)). As we move the str split into the `_repack_model.py`, we can clean up those condition check lines in `_utils.py`.